Mathlib/Data/Polynomial/Basic.lean

```diff
@@ -683,9 +683,12 @@ theorem coeff_zero (n : ℕ) : coeff (0 : R[X]) n = 0 :=
 #align polynomial.coeff_zero Polynomial.coeff_zero
 
 @[simp]
-theorem coeff_one_zero : coeff (1 : R[X]) 0 = 1 := by
-  rw [← monomial_zero_one, coeff_monomial]
-  simp
+theorem coeff_one {n : ℕ} : coeff (1 : R[X]) n = if n = 0 then 1 else 0 := by
```

Comment:
Making this a simp lemma seems a bit inconsistent with `coeff_X`, which is not a simp lemma. Generally simp lemmas that introduce `if`s can make the goal state unreadable if used too often.

Reply:
OK I've dropped this `simp`.